### PR TITLE
CNDB-11144 Allow to run on JDK22+ (update OSGI runtime)

### PR DESCRIPTION
**Pull request by jacek-lewandowski**
**What this PR does**:
- Update OSGI core from 6.0.0 to 8.0.0
- Update Apache Felix from 6.0.2 to 7.0.5
- Upgrade Mockito
- Allow BUILDING on JDK up to JDK23

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: coordinator/persistence-dse-next/pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
     <!-- Latest as of 2023-12-05: 4.0.11-b259738f492f -->
     <!-- Latest as of 2024-01-24: 4.0.11-0248d170a615 -->
     <!-- Latest as of 2024-04-29: 4.0.11-591d171ac9c9 -->
-    <cassandra.version>4.0.11-591d171ac9c9</cassandra.version>
+    <cassandra.version>4.0.11-5696964ff12e</cassandra.version>
```
Comment (tatu-at-datastax) [inline coordinator/persistence-dse-next/pom.xml:21]: When changing this version, please add a comment on date when update was made (to give an idea of how old DSE version is)

File: coordinator/pom.xml [hunk 2]
```diff
@@ -43,8 +43,8 @@
   <!-- Shared properties -->
   <properties>
     <!-- Compiler -->
-    <maven.compiler.source>11</maven.compiler.source>
-    <maven.compiler.target>11</maven.compiler.target>
+    <maven.compiler.source>17</maven.compiler.source>
```
Comment (eolivelli) [inline coordinator/pom.xml:46]: I don't think that we need to change this in order to make it run on JDK22
we can keep it built for JDK11, this way we reduce the risks of subtle behavior changes and also it won't change the developer enviroment setup for existing developers

WDYT ?

File: coordinator/persistence-dse-next/pom.xml (hunk 1 above)
Comment (eolivelli) [reply coordinator/persistence-dse-next/pom.xml:21]: do we need to update Cassandra ? IIUC we are only updating the dependencies in order to make it work on modern JDKs, the current version of Cassandra work decently on JDK22

File: .github/workflows/coordinator-test-v21.yml [hunk 3]
```diff
@@ -61,7 +61,7 @@ jobs:
         name: Setup Java JDK
         with:
           distribution: 'temurin'
-          java-version: 11
+          java-version: 17
```
Comment (eolivelli) [inline .github/workflows/coordinator-test-v21.yml:64]: I would keep this file and ensure that everything keeps working on jdk11 and create a second workload that builds on jdk22

**Review by eolivelli — COMMENTED** [review]
i have left a couple of comments, in general I think that it is better to keep supporting jdk11, this way it is easier for people to port changes/do hotfixes

File: coordinator/bridge/src/test/java/io/stargate/bridge/service/BaseBridgeServiceTest.java [hunk 4]
```diff
@@ -73,7 +72,7 @@ public class BaseBridgeServiceTest {
   protected @Mock Persistence persistence;
   protected @Mock AuthorizationService authorizationService;
 
-  protected Connection connection = spy(mock(Connection.class));
+  protected Connection connection = mock(Connection.class);
```
Comment (eolivelli) [inline coordinator/bridge/src/test/java/io/stargate/bridge/service/BaseBridgeServiceTest.java:75]: the new version of Mockito errored with "IllegalArgument Please don't pass mock here. Spy is not allowed on mock"

**Review by eolivelli — APPROVED** [review]

**Review by tatu-at-datastax — APPROVED** [review]

**Pull request merged**